src/pvt.c

```diff
@@ -295,21 +452,21 @@ s8 calc_PVT(const u8 n_used,
   double H[4][4];
 
   soln->valid = 0;
-
   soln->n_used = n_used; // Keep track of number of working channels
 
-  /* reset state to zero !? */
-  for(u8 i=4; i<8; i++) {
-    rx_state[i] = 0;
+  u8 removed_prn;
+  s8 flag = pvt_solve_and_check(rx_state, n_used, nav_meas, H, &removed_prn);
+
+  if (flag < 0) {
+    /* Didn't converge or least squares integrity check failed. */
+    log_warn("PVT iteration failed with code: %i\n", flag);
+    return -4;
   }
 
-  double update;
-  u8 iters;
-  /* Newton-Raphson iteration. */
-  for (iters=0; iters<PVT_MAX_ITERATIONS; iters++) {
-    if ((update = pvt_solve(rx_state, n_used, nav_meas, H)) > 0) {
-      break;
-    }
+  /* Initial solution failed, but repair was successful. */
+  if (flag == 1) {
+    log_info("pvt_repair successful. dropped prn: %i.\n", removed_prn);
+    soln->n_used--;
   }
```

Comment:
At the bottom of the function we should return something different if we have done a repair.